src/main/groovy/com/ableton/JenkinsMocks.groovy

```diff
@@ -165,18 +188,46 @@ class JenkinsMocks {
       throw new IllegalArgumentException('No mock output configured for script call ' +
         "'${script}', did you forget to call JenkinsMocks.addShMock()?")
     }
+
+    String stdout
+    int exitValue
+
+    // If the callback closure is not null, execute it and grab the output.
+    if (output.callback) {
+      Map callbackOutput = output.callback(script)
+      if (!callbackOutput || !(callbackOutput instanceof Map)) {
+        throw new IllegalArgumentException("Mocked sh callback for ${script} did not" +
+          ' return expected output type')
+      }
+      if (!callbackOutput.containsKey('stdout')
+        || !(callbackOutput['stdout'] instanceof String)) {
+        throw new IllegalArgumentException("Mocked sh callback for ${script} did not" +
```

Comment:
We should have a test for this failure mode. I don't see one, but maybe I'm missing something.